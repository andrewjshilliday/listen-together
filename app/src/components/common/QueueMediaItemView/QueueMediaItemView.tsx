import React from 'react';
import { useMusicKit } from '../../providers';
import { MusicKitService } from '../../../services';
import styled from 'styled-components';

interface QueueMediaItemViewProps {
  item: MusicKit.MediaItem,
  index: number
}

const QueueMediaItemView: React.FC<QueueMediaItemViewProps> = ({ item, index }) => {
  const musicKitProvider = useMusicKit();
  
  return (
    <QueueItem>
      <img src={MusicKitService.FormatArtwork(item.attributes.artwork, 70)} className="img-fluid rounded" alt={item.attributes.name}/>
      <QueueItemText nowPlaying={musicKitProvider.musicKit.player.queue.position === index}>
        <span className="text-truncate">{item.attributes.name}</span>
        <span className="text-truncate">{item.attributes.albumName}</span>
        <span className="text-truncate">{item.attributes.artistName}</span>
      </QueueItemText>
    </QueueItem>
  );
}

export default QueueMediaItemView;


const QueueItem = styled.div`
  display: flex;
  align-items: center;
  height: 80px;
  width: 100%;
  padding: 0 15px;
  
  img {
    height: 70px;
    width: 70px;
  }
`;
const QueueItemText = styled.div<{nowPlaying: boolean}>`
  margin-left: 20px;
  overflow: hidden;
  span { display: block; }
  ${props => props.nowPlaying && `
    color: var(--primary);
  `}
`;
