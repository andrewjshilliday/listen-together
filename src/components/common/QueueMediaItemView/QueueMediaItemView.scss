.queue-item {
  display: flex;
  align-items: center;
  height: 80px;
  width: 100%;
  padding: 0 15px;

  .queue-item-text {
    margin-left: 20px;
    overflow: hidden;
    span { display: block; }
  }
  
  img {
    height: 70px;
    width: 70px;
  }
}

.now-playing {
  color: red;
}
