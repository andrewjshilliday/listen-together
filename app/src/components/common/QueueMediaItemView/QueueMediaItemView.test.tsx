import React from 'react';
import renderer from 'react-test-renderer';
import QueueMediaItemView from './QueueMediaItemView';

jest.mock('../../../services/MusicKit');
jest.mock('../../providers/MusicKitProvider', () => require('../../../__mocks__/providers/MusicKitProvider.mock').default);

it('renders correctly', () => {
  const tree = renderer.create(<QueueMediaItemView item={require('../../../__mocks__/objects/Album.mock').default} index={0} />).toJSON();
  expect(tree).toMatchSnapshot();
});
