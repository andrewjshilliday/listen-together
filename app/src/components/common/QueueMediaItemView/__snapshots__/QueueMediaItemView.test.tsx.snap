// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="sc-AxjAm GaNVe"
>
  <img
    alt="This Land"
    className="img-fluid rounded"
  />
  <div
    className="sc-AxirZ iXMdxL"
  >
    <span
      className="text-truncate"
    >
      This Land
    </span>
    <span
      className="text-truncate"
    />
    <span
      className="text-truncate"
    >
      Gary Clark Jr.
    </span>
  </div>
</div>
`;
